client/SMCE_Client.cpp

```diff
@@ -0,0 +1,339 @@
+        t_lock.unlock();                                                                                        
+        if(len == 0){                                                                                                            
+            std::this_thread::sleep_for(1ms);                                                                                    
+            continue;                                                                                                            
+        }                                                                                                                        
+        buffer.resize(len);
+        if(file_write){
+            uart_to_file(buffer,file);
+        }else{
+            if(!mute_uart){
+                std::cout << termcolor::red << buffer << termcolor::reset <<std::endl << "$>" << std::flush; 
+            }  
+        }                                                                                                                                                                             
+    }                                                                                                                            
+}                                                                                                                                
+// Prints a command menu for SMCE_Client                                                                                         
+void print_menu(){                                                                                                               
+    std::cout << "SMCE Client menu:" << std::endl;                                                                               
+    std::cout << "-h -> See menu" << std::endl;                                                                                  
+    std::cout << "-p -> Pause or resume the board" << std::endl;                                                                 
+    std::cout << "-rc -> Recompile the sketch" << std::endl;                                                                     
+    std::cout << "-m <message> -> Send message to board through uart (serial)" << std::endl;
+    std::cout << "-mt -> disable uart prints in console" << std::endl;                                     
+    std::cout << "-wa <pin> <value> -> Set a specific value on a analog pin" << std::endl;                                       
+    std::cout << "-wd <pin> <value> -> Set a specific value on a digital pin, value should be 0 or 1" << std::endl;              
+    std::cout << "-ra <pin> -> Read the value on a analog pin" << std::endl;                                                     
+    std::cout << "-rd <pin> -> Read the value on a digital pin" << std::endl;                                                    
+    std::cout << "-q -> Power off board and quit program" << std::endl << std::endl;                                             
```

Comment:
All these lines could be a single expression, with a single, initial `std::cout`. Also all those `std::endl`s might as well be newline characters (`\n`) in the preceding string. Note that adjacent C++ string literals are concatenated by the preprocessor.